data/src/main/java/com/teamagam/gimelgimel/data/geometry/repository/GeoDataRepository.java

```diff
@@ -0,0 +1,72 @@
+package com.teamagam.gimelgimel.data.geometry.repository;
+
+import com.teamagam.gimelgimel.data.geometry.repository.inMemory.InMemoryGeoDataCache;
+import com.teamagam.gimelgimel.domain.geometries.entities.GeoEntity;
+import com.teamagam.gimelgimel.domain.geometries.entities.VectorLayer;
+import com.teamagam.gimelgimel.domain.geometries.repository.GeoRepository;
+
+import rx.Observable;
+
+public class GeoDataRepository implements GeoRepository {
+
+    private InMemoryGeoDataCache mCache;
+
+    public GeoDataRepository(InMemoryGeoDataCache cache) {
+        mCache = cache;
+    }
+
+    @Override
+    public Observable<VectorLayer> getAllVectorLayers() {
+        return mCache.getAllVectorLayers();
+    }
+
+    @Override
+    public Observable<VectorLayer> getVectorLayerById(String id) {
+        return mCache.getVectorLayerById(id);
+    }
+
+    @Override
+    public Observable<GeoEntity> getEntity(String layerId, String entityId) {
+        return mCache.getEntity(layerId, entityId);
+    }
+
+    @Override
+    public void addVectorLayer(VectorLayer vectorLayer) {
+        mCache.addVectorLayer(vectorLayer);
+    }
+
+    @Override
+    public void addVectorLayers(VectorLayer[] vectorLayers) {
```

Comment:
Arrays are more generic way, because we can convert every data type of list to array.
I can change it to list because it's interface.